### PR TITLE
test: add ut for hertz/pkg/common/adaptor

### LINKED ISSUE #257: Request help: Supplementary of Unit Tests
> We hope more developers join CloudWeGo. This is a task issue that needs your help.

**Issue Description**
Write Unit Tests for some modules of Hertz, we mainly focus on **Statement Coverage**. The packages are shown below, you can choose and reply by comment.
Before you pick the task, please comply with the following requirements:
- The Unit Test you add must be meaningful. 
  - It is used to verify the correctness of code logic.
  - It is forbidden to add a test only for the purpose of coverage but without logical meaning.
- It is suggested to add comments with clear descriptions for each unit test.
- Every unit test must have assertions.  
- Don't check code logic by `print`.

**Packages**
> Current Coverages v0.3.1.
---
- [ ] `pkg/app`: Current  70.87%,  Target **80%**, by @chaoranz758 
- [ ] `pkg/route`: Current 78.49%,  Target **85%**, by 
- [x] `pkg/protocol/http1`: Current 33.94%,  Target **80%**, by @a631807682 
- [x] `pkg/protocol/uri.go`: Current 70.35%,  Target **80%**, Raised to 86.4%, by @L2ncE
- [x] `pkg/protocol/args.go`: Current 56.57%,  Target **80%**, Raised to 81.3%, by @justlorain
- [x] `pkg/protocol/header.go`: Current 74.25%,  Target **80%**, by @XiaoYi-byte 
- [x] `pkg/protocol/request.go`: Current 51.31%,  Target **80%**, Raised to 85.3%, by @BaiZe1998 
- [x] `pkg/protocol/response.go`: Current 60.14%(Overall 56%),  Target **80%**,  Raised to 81.8% by @gityh2021 
- [x] `pkg/protocol/multipart.go`: Current 20.25%,  Target **80%**, Raised to 79.75%,  by @gityh2021
- [x] `pkg/network`: Current 68.14%,  Target **80%**, Raised to 95.2%,    by @a631807682
- [ ] `pkg/common/utils`: Current  72.34%,  Target **80%**,  by 
- [x] `pkg/common/config`: Current 32.91%,  Target **80%**, Raised to 100%, by @rogerogers 
- [x] `pkg/common/adaptor`: Current 76.60%,  Target **85%**, Raised to 91.5%, by @L2ncE
- [x] `internal/bytesconv`: Current 1.68%,  Target **80%**, Raised to 82.35%, by @Alexdener
- [x]  `pkg/common/test`: Current 3.2%,  Target **80%**,  Raised to 98.1% by @a631807682 

Standard coverage stats: https://app.codecov.io/gh/cloudwego/hertz

### PR DESCRIPTION
**Pull request by L2ncE**
#### What type of PR is this?
test

#### Check the PR title.

- [x] This PR title match the format: \<type\>(optional scope): \<description\>
- [x] The description of this PR title is user-oriented and clear enough for others to understand.

#### (Optional) More detail description for this PR(en: English/zh: Chinese).

en: add ut for `hertz/pkg/common/adaptor`, coverage increased from 76.6% to 92.3%
zh(optional): 增加对`hertz/pkg/common/adaptor`的单测，覆盖率从76.6%提高到了92.3%

#### Which issue(s) this PR fixes:

#257

### CODE REVIEW

**Comment by li-jin-gou** [comment]
It is forbidden to add a test only for the purpose of coverage but without logical meaning.

File: pkg/common/adaptor/request_test.go [hunk 1]
```diff
@@ -38,6 +39,7 @@ func TestCompatResponse_WriteHeader(t *testing.T) {
 	testHeader["Key1"] = []string{"value1"}
 	testHeader["Key2"] = []string{"value2", "value22"}
 	testHeader["Key3"] = []string{"value3", "value33", "value333"}
+	testHeader[consts.HeaderSetCookie] = []string{"cookie"}
```
Comment (li-jin-gou) [inline pkg/common/adaptor/request_test.go:42]: What is the purpose of adding this key?

File: pkg/common/adaptor/request_test.go [hunk 2]
```diff
@@ -109,6 +111,12 @@ func handlerAndCheck(t *testing.T, writer http.ResponseWriter, request *http.Req
 	for k, v := range reqHeader {
 		respHeader[k] = v
 	}
+
+	_, err = writer.Write(nil)
```
Comment (li-jin-gou) [inline pkg/common/adaptor/request_test.go:115]: why use writer to write `nil`?

File: pkg/common/adaptor/request_test.go (hunk 1 above)
Comment (L2ncE) [reply pkg/common/adaptor/request_test.go:42]: > What is the purpose of adding this key?

为了测试这段代码能否正确执行
```go
if k == consts.HeaderSetCookie {
    cookie := protocol.AcquireCookie()
    cookie.Parse(vv)
    c.h.Header.SetCookie(cookie)
    continue
}
```
我没有在这里测试`SetCookie`是否有效是因为它已经在`pkg/protocol/header_test.go`中进行了测试

File: pkg/common/adaptor/request_test.go (hunk 2 above)
Comment (L2ncE) [reply pkg/common/adaptor/request_test.go:115]: > why use writer to write `nil`?

在这里 write `nil`时，由于`!c.writeHeader`为 true，会执行`c.WriteHeader(consts.StatusOK)`，此时的`expectStatusCode`应为200，而后又执行了`writer.WriteHeader(statusCode)`，使`expectStatusCode`重新设置为了299使测试通过

File: pkg/common/adaptor/request_test.go (hunk 1 above)
Comment (li-jin-gou) [reply pkg/common/adaptor/request_test.go:42]: How to determine if it is normal? this seems to only allow the code to execute up to here and we should use `assert` in unit test to verify that it is correct.

File: pkg/common/adaptor/request_test.go (hunk 2 above)
Comment (li-jin-gou) [reply pkg/common/adaptor/request_test.go:115]: Ibid

File: pkg/common/adaptor/request_test.go (hunk 1 above)
Comment (L2ncE) [reply pkg/common/adaptor/request_test.go]: > How to determine if it is normal? this seems to only allow the code to execute up to here and we should use `assert` in unit test to verify that it is correct.

Done. Parse the cookie and use `assert.DeepEqual` to verify correctness

File: pkg/common/adaptor/request_test.go (hunk 2 above)
Comment (L2ncE) [reply pkg/common/adaptor/request_test.go]: > Ibid

Done, Use an extra `makeACall` to verify the status code case without using `WriteHeader`

File: pkg/common/adaptor/request_test.go [hunk 3]
```diff
@@ -77,16 +90,16 @@ func makeACall(t *testing.T, method, url string, header http.Header, body string
...
-		t.Fatalf("Status code not equal: want: %d, got: %d", expectStatusCode, resp.StatusCode)
-	}
+	// Parse out the cookie to verify it is correct
+	cookie := protocol.Cookie{}
+	_ = cookie.Parse(header[consts.HeaderSetCookie][0])
+	assert.DeepEqual(t, expectCookieValue, cookie.Value())
 }
 
-func handlerAndCheck(t *testing.T, writer http.ResponseWriter, request *http.Request, wantHeader http.Header, wantBody string, statusCode int) {
+func handlerAndCheck(t *testing.T, writer http.ResponseWriter, request *http.Request, wantHeader http.Header, wantBody string, statusCode ...int) {
```
Comment (welkeyever) [inline pkg/common/adaptor/request_test.go:102]: What is the purpose of this change: `statusCode` int -> ...int?

Comment (L2ncE) [reply pkg/common/adaptor/request_test.go:102]: > What is the purpose of this change: `statusCode` int -> ...int?

Changed to make this parameter optional. When testing the code before this change, a meaningless number 0 will be passed in and judged in the subsequent logic. I think this change can make the code more elegant

Before
```go
handlerAndCheck(t, resp, req, testHeader, testBody, 0)
```
Now
```go
handlerAndCheck(t, resp, req, testHeader, testBody)
```

Before
```go
if statusCode != 0 {
    writer.WriteHeader(statusCode)
}
```

Now
```go
if statusCode != nil {
    writer.WriteHeader(statusCode[0])
}
```

Comment (welkeyever) [reply pkg/common/adaptor/request_test.go:102]: From a function signature perspective, it's a little bit strange, maybe add some comment to introduce?

Comment (L2ncE) [reply pkg/common/adaptor/request_test.go:107]: > From a function signature perspective, it's a little bit strange, maybe add some comment to introduce?

Done.

**Review by welkeyever — APPROVED** [review]
Thanks！lgtm

**Review by li-jin-gou — APPROVED** [review]
